bootstrap/util/configowner.go

```diff
@@ -125,6 +127,16 @@ func (co ConfigOwner) KubernetesVersion() string {
 
 // GetConfigOwner returns the Unstructured object owning the current resource.
```

Comment:
I would add a warning that this can impact performance at scale because it reads unstructured objects which are not cached

Reply:
Should we deprecate this func?

Reply:
I would say not without knowing why me made this public, I assume because someone asked for it from providers

Reply:
Looks like there is one usage in the CAPA controller: https://cs.k8s.io/?q=GetConfigOwner&i=nope&files=&excludeFiles=&repos=

I don't know if they intentionally use the version which triggers an uncached call.

But definitely fine for me to not deprecate in this PR to get the improvement in ASAP.